sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapData.scala

```diff
@@ -26,6 +26,9 @@ import org.apache.spark.util.collection.Utils
  *
  * Note that, user is responsible to guarantee that the key array does not have duplicated
  * elements, otherwise the behavior is undefined.
+ *
+ * @throws IllegalArgumentException If the number of elements in `keyArray` and `valueArray`
+ *                                  are different.
  */
 class ArrayBasedMapData(val keyArray: ArrayData, val valueArray: ArrayData) extends MapData {
   require(keyArray.numElements() == valueArray.numElements())
```

Comment:
I have to leave it as is because JSON datasource expects `IllegalArgumentException` otherwise it propagates `SparkException` w/ `INTERNAL_ERROR` to users.